src/components/widgets/accounts-balance-sheet/accounts-balance-sheet.directive.coffee

```diff
@@ -100,20 +100,20 @@ module.controller('WidgetAccountsBalanceSheetCtrl', ($scope, $q, ImpacWidgetsSvc
         return true
 
   translateCategories = (categories) ->
-    _.each(categories, (category) ->
+    translationPromises = _.map(categories, (category) ->
       $translate('impac.widget.account_balance_sheets.' + category.toLowerCase()).then(
         (translation) ->
-          $scope.categories.push({label: translation, key: category})
+          {label: translation, key: category}
 
-        (translationId) ->  # If there is no translation, keep the original
-          $scope.categories.push({label: category.toLowerCase(), key: category})
+        (translationId) ->
+          {label: category.toLowerCase(), key: category}
+        )
       )
-      return
-    )
+    Promise.all(translationPromises)
```

Comment:
Probs best to continue using `$q` for promises instead of the JS `Promise`. As it would be the first use in impac-angular. 
```
$q.all(translationPromises)
```

The behaviour will be the same.